lib/optimizely/index.js

```diff
@@ -142,11 +142,19 @@ Optimizely.prototype.activate = function(experimentKey, userId, attributes) {
                                                       impressionEvent.url,
                                                       JSON.stringify(impressionEvent.params));
     this.logger.log(LOG_LEVEL.DEBUG, dispatchedImpressionEventLogMessage);
-    this.eventDispatcher.dispatchEvent(impressionEvent)
-      .then(function() {
+
+    var eventDispatcherCallback = function() {
         var activatedLogMessage = sprintf(LOG_MESSAGES.ACTIVATE_USER, MODULE_NAME, userId, experimentKey);
         this.logger.log(LOG_LEVEL.INFO, activatedLogMessage);
-      }.bind(this));
+    }.bind(this)
+
+     var eventDispatcherResponse = this.eventDispatcher.dispatchEvent(impressionEvent, eventDispatcherCallback);
```

Comment:
Since this is used in two places, can you move it the shared logic into a private function in the same file:

```function dispatchEvent(event, callback) {
}```